src/WinRT.Runtime/ComWrappersSupport.cs

```diff
@@ -315,6 +315,19 @@ private static bool IsIReferenceArray(Type implementationType)
                 _boxedValueReferenceCache.Add(resultingObject, inspectable);
                 return resultingObject;
             };
+        }
+
+        private static Func<IInspectable, object> CreateCustomTypeMappingFactory(Type customTypeHelperType)
+        {
+            var fromAbiMethod = customTypeHelperType.GetMethod("FromAbi", BindingFlags.Public | BindingFlags.NonPublic | BindingFlags.Static);
+            if (fromAbiMethod is null)
+            {
+                throw new MissingMethodException();
+            }
+
+            var parms = new[] { Expression.Parameter(typeof(IInspectable), "obj") };
```

Comment:
if you're looking to reduce allocations, this could be shared (global) and combined with a few other places in ComWrappers code

Reply:
These methods are cached but it can still end up overall reducing memory allocations.  I will look at evaluating the perf diff and address that in a future PR if I see differences.